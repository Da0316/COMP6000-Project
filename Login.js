import { StatusBar } from "expo-status-bar";
import { useState } from "react";
import {
  StyleSheet,
  Text,
  View,
  Image,
  TextInput,
  Button,
  TouchableOpacity,
} from "react-native";

export default function Login() {
  const [userName,setuserName] =useState('');
  const [password,setpassword] =useState('');
  return (
    <View style={styles.container}>
      <Text style={styles.title}>Login Page</Text>
      <StatusBar style="auto" />
      <View style={styles.inputView}>
        <TextInput
          style={styles.TextInput}
          placeholder="userName"
          placeholderTextColor={'#3c3744'}
          onChangeText={(userName) => setzuserName(userName)}
        />
      </View>

      <View style={styles.inputView}>
      <TextInput
          style={styles.TextInput}
          placeholder="Password"
          placeholderTextColor={'#3c3744'}
          onChangeText={(password) => setpassword(password)}
        />
      </View>

      <TouchableOpacity >
        <Text style={styles.forgotPassBtn}>Forgot Password?</Text>
      </TouchableOpacity>

      <TouchableOpacity style={styles.LoginBtn}>
        <Text style={styles.LoginText}>Login</Text>
      </TouchableOpacity>

    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#EEF5DB',
    alignItems: 'center',
    justifyContent: 'center',
  },
  title:{
    padding:20,
    fontSize: 25,
    fontFamily:"Cochin",
    marginBottom:50
  },
  inputView:{
    backgroundColor: '#B8D8D8',
    borderRadius:30,
    width:"70%",
    height:45,
    marginBottom:20,
    alignItems:"center"
  },
  TextInput:{
    height:50,
    flex:1,
    padding:10,
    //marginRight:10,
    alignItems:'center'
  },
  forgotPassBtn:{
    height:30,
    marginBottom:10
  },
  LoginBtn:{
    width: "70%",
    borderRadius: 25,
    height: 50,
    alignItems: "center",
    justifyContent: "center",
    marginTop: 20,
    backgroundColor: "#FE5F55",
  }

});
