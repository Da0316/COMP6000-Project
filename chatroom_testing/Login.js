import React from 'react';
import {Button, StyleSheet, TextInput, View} from 'react-native';

export default function Login({onLogin, username, setUsername}) {
  return (
    <View style={styles.container}>
      <TextInput
        style={styles.input}
        onChangeText={setUsername}
        value={username}
      />
      <Button title={'Login'} onPress={onLogin} />
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    backgroundColor: '#cacaca',
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  input: {
    backgroundColor: 'white',
    width: '80%',
    marginBottom: 20,
    borderRadius: 20,
    paddingHorizontal: 20,
  },
});