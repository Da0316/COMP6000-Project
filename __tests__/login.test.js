import React from "react";
// import renderer from "react-test-renderer";
import {
  render,
  fireEvent,
  waitFor,
} from "@testing-library/react-native";

// Import the component to be tested
import Login from "../screens/login";
import AsyncStorage from "@react-native-async-storage/async-storage";

jest.mock("@react-native-async-storage/async-storage", () => ({
  setItem: jest.fn(() => Promise.resolve()),
}));

// Mock the useIsFocused hook
jest.mock("@react-navigation/native", () => ({
  useIsFocused: jest.fn(),
}));

test("render default elements", () => {
  const { getAllByText, getByPlaceholderText } = render(<Login />);

  expect(getAllByText("Login").length).toBe(1);
  expect(getAllByText("Signup").length).toBe(1);
  expect(getAllByText("Forgot Password? Press here").length).toBe(1);

  getByPlaceholderText("Username");
  getByPlaceholderText("Password");
});

test("username blank", () => {
  const { getByTestId } = render(<Login />);
  const alertMock = jest.fn();
  window.alert = alertMock;

  fireEvent.press(getByTestId("loginButton"));

  expect(alertMock).toHaveBeenCalledWith("Please enter username");

  window.alert.mockRestore();
});

test("password blank", () => {
  const { getByTestId } = render(<Login />);
  const alertMock = jest.fn();
  window.alert = alertMock;

  fireEvent.changeText(getByTestId("usernameTest"), "test");
  fireEvent.press(getByTestId("loginButton"));

  expect(alertMock).toHaveBeenCalledWith("Please enter a password");

  window.alert.mockRestore();
});

test("incorrect details", async () => {
  const { getByTestId } = render(<Login />);
  const alertMock = jest.fn();
  window.alert = alertMock;
  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve(-1),
    })
  );
  fireEvent.changeText(getByTestId("usernameTest"), "testuser");
  fireEvent.changeText(getByTestId("passwordTest"), "testpassword");
  fireEvent.press(getByTestId("loginButton"));
  await waitFor(() =>
    expect(alertMock).toHaveBeenCalledWith("Wrong Login Details")
  );
  window.alert.mockRestore();
});

test ("correct details & async storage", async () => {
  const { getByTestId } = render(<Login />);
  const alertMock = jest.fn();
  window.alert = alertMock;
  global.fetch = jest.fn(() =>
    Promise.resolve({
      json: () => Promise.resolve("1"),
    })
  );
  fireEvent.changeText(getByTestId("usernameTest"), "lkaasan");
  fireEvent.changeText(getByTestId("passwordTest"), "hello");
  fireEvent.press(getByTestId("loginButton"));
  await waitFor(() =>
    expect(alertMock).toHaveBeenCalledWith("Successfully Login"),
  );

  expect(AsyncStorage.setItem).toHaveBeenCalledWith(
    "user_id",
    JSON.stringify(1)
  );

  window.alert.mockRestore();
})
