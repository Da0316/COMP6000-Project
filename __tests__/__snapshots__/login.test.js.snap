// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> renders correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#000000",
        "flex": 2,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://raptor.kent.ac.uk/proj/comp6000/project/08/images/O.png",
        }
      }
      style={
        {
          "alignSelf": "center",
          "flex": 1,
          "height": "80%",
          "justifyContent": "flex-end",
          "position": "absolute",
          "resizeMode": "cover",
          "width": "40%",
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 80,
        "borderTopRightRadius": 80,
        "flex": 6,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#EBEBEB",
          "fontSize": 25,
          "letterSpacing": 7,
          "position": "absolute",
        }
      }
    >
      Welcome
    </Text>
    <Text
      style={
        {
          "color": "FFFFFF",
          "fontSize": 25,
          "letterSpacing": 7,
          "marginBottom": 10,
          "padding": 20,
        }
      }
    >
      Welcome
    </Text>
    <Text
      style={
        [
          {
            "color": "rgba(28, 27, 31, 1)",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 0.54)",
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 12,
                "letterSpacing": 0.4,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              {
                "marginBottom": 8,
              },
            ],
          ],
        ]
      }
    >
      Login with your username and password
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#EBEBEB",
          "borderRadius": 30,
          "height": 45,
          "marginBottom": 10,
          "width": "70%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Username"
        placeholderTextColor="#3c3744"
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "height": 50,
            "padding": 10,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#EBEBEB",
          "borderRadius": 30,
          "height": 45,
          "marginBottom": 10,
          "width": "70%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#3c3744"
        secureTextEntry={true}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "height": 50,
            "padding": 10,
          }
        }
        value=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1A1918",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 1,
          "opacity": 1,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "height": 30,
            "marginBottom": 10,
            "marginTop": 5,
          }
        }
      >
        Forgot Password?
      </Text>
    </View>
    <Text>
      Don't have an account?
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f9ce40",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 1,
          "opacity": 1,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        Signup
      </Text>
    </View>
  </View>
</View>
`;

exports[`snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#000000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "#000000",
        "flex": 2,
        "height": "100%",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://raptor.kent.ac.uk/proj/comp6000/project/08/images/O.png",
        }
      }
      style={
        {
          "alignSelf": "center",
          "flex": 1,
          "height": "80%",
          "justifyContent": "flex-end",
          "position": "absolute",
          "resizeMode": "cover",
          "width": "40%",
        }
      }
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderTopLeftRadius": 80,
        "borderTopRightRadius": 80,
        "flex": 6,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "#EBEBEB",
          "fontSize": 25,
          "letterSpacing": 7,
          "position": "absolute",
        }
      }
    >
      Welcome
    </Text>
    <Text
      style={
        {
          "color": "FFFFFF",
          "fontSize": 25,
          "letterSpacing": 7,
          "marginBottom": 10,
          "padding": 20,
        }
      }
    >
      Welcome
    </Text>
    <Text
      style={
        [
          {
            "color": "rgba(28, 27, 31, 1)",
          },
          {
            "textAlign": "left",
          },
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 0.54)",
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 12,
                "letterSpacing": 0.4,
                "lineHeight": 20,
                "marginVertical": 2,
              },
              {
                "marginBottom": 8,
              },
            ],
          ],
        ]
      }
    >
      Login with your username and password
    </Text>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#EBEBEB",
          "borderRadius": 30,
          "height": 45,
          "marginBottom": 10,
          "width": "70%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Username"
        placeholderTextColor="#3c3744"
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "height": 50,
            "padding": 10,
          }
        }
        value=""
      />
    </View>
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#EBEBEB",
          "borderRadius": 30,
          "height": 45,
          "marginBottom": 10,
          "width": "70%",
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Password"
        placeholderTextColor="#3c3744"
        secureTextEntry={true}
        style={
          {
            "alignItems": "center",
            "flex": 1,
            "height": 50,
            "padding": 10,
          }
        }
        value=""
      />
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#1A1918",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 1,
          "opacity": 1,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        Login
      </Text>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
    >
      <Text
        style={
          {
            "height": 30,
            "marginBottom": 10,
            "marginTop": 5,
          }
        }
      >
        Forgot Password?
      </Text>
    </View>
    <Text>
      Don't have an account?
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f9ce40",
          "borderRadius": 25,
          "height": 50,
          "justifyContent": "center",
          "marginTop": 1,
          "opacity": 1,
          "width": "70%",
        }
      }
    >
      <Text
        style={
          {
            "color": "#fff",
            "fontWeight": "bold",
          }
        }
      >
        Signup
      </Text>
    </View>
  </View>
</View>
`;
